nearcore/src/config.rs

```diff
@@ -303,7 +302,6 @@ pub struct Config {
     pub tracked_accounts: Vec<AccountId>,
     pub tracked_shards: Vec<ShardId>,
     pub archive: bool,
-    pub log_summary_style: LogSummaryStyle,
```

Comment:
We actually don’t want to that just yet.  Removing the field will cause a warning about unknown field and that will lead to confusion amongst users.  Ideally, we would have a period when the field being present would give a warning with instructions how to deal with it (rather than generic ‘unknown field’ warning).

To do that I’d suggest first adding a default variant to `LogSummaryStyle`:

```rust
#[derive(Default)]
pub enum LogSummaryStyle {
    #[default]
    Unset,
    #[serde(rename = "plain")]
    Plain,
    #[serde(rename = "colored")]
    Colored,
}

impl LogSummaryStyle {
    const fn is_unset(&self) -> bool {
        matches!(Self::Unset, self)
    }
}
```

Then changing the field in Config to use default if missing:

```rust
pub struct Config {
    /* ... */
    #[serde(default, skip_serializing_if = "LogSummaryStyle::is_unset")]
    pub log_summary_style: LogSummaryStyle,
    /* ... */
}
```

along with changing any other defaults to use Unsest.  And then, when reading config from file (in from_file method), if log_summary_style field’s value is other than Unset, issue warning explaining that the field is no longer used and users should remove it from their config.json.